### PR TITLE
[Logic App Integration Account][Bug Fix] : Swagger spec updated as per the latest changes in the RP …

### PR DESCRIPTION
**Pull request by pankajsn**
…definitions.

Issue fixed: As the LogicApps and Integration Account are in GA, swagger spec is merged
Issue fixed: HashingAlgorithm enum in Azure SDK swagger spec is not updated with new values.
Issue fixed: Not able to create New-AzureRmIntegrationAccountAgreement with envelopeOverrides/responsibleAgencyCode due to RP definition updates.
Issue fixed: Integration Account Sku name values fixed as per review comments.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by pankajsn** [comment]
@jhendrixMSFT  I am seeing 110 validation errors regarding property description in the spec. We are not describing the while defining the object but while referring. Please let us know if we need to fix this. Our current change is only few bug fixes.

**Comment by jhendrixMSFT** [comment]
Hi @pankajsn I'm following up on your question and should have an answer for soon.

**Comment by jhendrixMSFT** [comment]
@pankajsn sorry for the delay.  We'll need these validation errors to be fixed, the reason being that they're used for generating docs during SDK generation.

**Comment by pankajsn** [comment]
@jhendrixMSFT validation errors are fixed.

**Comment by pankajsn** [comment]
Hi Joel,

I have fixed the validation warnings.
Please let us know if there is any other pending action item on us. We have respective SDK and PowerShell updates pending on this, that we are targeting to release with February milestone.

Regards,
~Pankaj

File: arm-logic/2016-06-01/swagger/logic.json
Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Breaking change on several counts.
1. New value is added to the list
2. New value is added at the top of the list, instead of it being in the bottom. Previous version had Primary and Secondary.

**Comment by amarzavery** [comment]
@pankajsn - We have hooked up a bunch of validation tools as a part of our CI .

The linter shows some warnings. Please make sure you address them before the PR can be merged.
Warnings can be found [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/199003272#L2326).
```
  AutoRest Linter validation:
Executing: mono ./AutoRest.*/tools/AutoRest.exe -CodeGenerator None -I arm-logic/2016-06-01/swagger/logic.json
WARNING: NonAppJsonTypeWarning - Media types other than 'application/json' has limited support
	Path: #/Consumes[2]
WARNING: NonAppJsonTypeWarning - Media types other than 'application/json' has limited support
	Path: #/Produces[2]
WARNING: OperationsAPIImplementationValidation - Operations API must be implemented for the service.
	Path: #/Paths
WARNING: ResourceModelValidation - The id, name, type, location and tags properties of the Resource must be present with id, name and type as read-only
	Path: #/Definitions
WARNING: TrackedResourceValidation - Tracked Resource failing validation is: Workflow. Validation Failed: 4. 
    A Tracked Resource must have: 
    1. A Get Operation 
    2. A ListByResourceGroup operation with x-ms-pageable extension and 
    3. A ListBySubscriptionId operation with x-ms-pageable extension.
    4. Type, Location, Tags should not be used in the properties.
	Path: #/Definitions
WARNING: SkuModelValidation - Sku Model is not valid. A Sku model must have name property. It can also have tier, size, family, capacity as optional properties.
	Path: #/Definitions
```

**Comment by amarzavery** [comment]
Hi @pankajsn,

Can you please provide examples for request and response using the extension x-ms-examples? This will 
- help us validate the examples against the swagger spec and 
- it will also form the basis for customer facing REST API documentation on docs.microsoft.com website.

You can more info about the extension and an example swagger spec over [here](https://github.com/Azure/adx-documentation-pr/blob/master/tools/tools.md#swagger-model-validation).

File: arm-logic/2016-06-01/swagger/logic.json
Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: @pankajsn There is an issue https://github.com/Azure/azure-rest-api-specs/issues/898 reported by a customer which is _correct_. Can you please resolve this issue, by accurately representing the properties of "Resource" `"readOnly": true`.

- "id", "name" and "type" should be marked `"readOnly": true` as per ARM guidelines.
- Shouldn't "location" be a required property `"required": ["location"]`?

**Comment by pankajsn** [comment]
@Amar Zavery<mailto:amzavery@microsoft.com>
These are little confusing or not accurate. Please find CIL and help us understanding these warning.

**Comment by jhendrixMSFT** [comment]
I sat down with @amarzavery regarding your questions, below are the answers.  Let me know if you have further questions.

NonAppJsonTypeWarning - can you please verify that your RP actually consumes and produces text/json in addition to application/json?

OperationsAPIImplementationValidation - you need to implement the operations API.  For an example please see https://github.com/Azure/azure-rest-api-specs/blob/master/arm-cdn/2016-10-02/swagger/cdn.json#L1462

TrackedResourceValidation - the output here is unclear and we will work to fix that.  Only item #4 (type, location etc) has an issue (that's what the "Validation Failed: 4" means) and this appears to be a bug in the validator.  You can ignore this for now.

SkuModelValidation - another unclear error message which we'll address.  The "name" field needs to be marked as required.

**Comment by pankajsn** [comment]
@Joel Hendrix<mailto:jhendrix@microsoft.com> @Amar Zavery<mailto:amzavery@microsoft.com>

I have fixed below warnings (except NonAppJsonTypeWarning: I am following up on that). I hope this is not blocking this PR as this change is not introduced as part of this PR. Once confirmed, I will start a new PR for that.

**Comment by pankajsn** [comment]
RP does not support text/json. Updated the spec. Please review the PR.

From: Pankaj Singh Negi
Sent: Tuesday, February 7, 2017 4:47 PM
To: 'Azure/azure-rest-api-specs' <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>; Joel Hendrix <jhendrix@microsoft.com>; Amar Zavery <amzavery@microsoft.com>
Cc: Mention <mention@noreply.github.com>; Vinay Singh <vinaysin@microsoft.com>
Subject: RE: [Azure/azure-rest-api-specs] [Logic App Integration Account][Bug Fix] : Swagger spec updated as per the latest changes in the RP … (#891)

@Joel Hendrix<mailto:jhendrix@microsoft.com> @Amar Zavery<mailto:amzavery@microsoft.com>

I have fixed below warnings (except NonAppJsonTypeWarning: I am following up on that). I hope this is not blocking this PR as this change is not introduced as part of this PR. Once confirmed, I will start a new PR for that.

**Review by amarzavery — APPROVED** [review]

File: arm-logic/2016-06-01/swagger/logic.json
Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: The name of the operation is List where as the return type of the object is a string. The customer would expect that the service would return an array or an object that has some property that is of type array. Please fix the name of the method to _GetCallbackUrl

File: arm-logic/2016-06-01/swagger/logic.json
Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Is this a string or an array of strings?
See if you would have given x-ms-examples, our job would have been easy and the CI would have caught if something is wrong.

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: allOf on the Resource should be fine.

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: What is the schema of content. How useful are free form objects to customers?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Are there any required properties over here? What do you expect the customer to send bare minimum while creating (PUT) on the "IntegrationAccountSchema"

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: should this be an enum?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: missing description

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: enum? How would the customer know what to put over here while creating the Map? How important is the usability of the API? REST API documentation will be generated from swagger. The description of the properties are not that helpful. Where do you expect the customer to  go to find out what can he/she provide over here?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: missing description

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: what are the required properties over here?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: required properties?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: enum?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: any required or readOnly properties over here?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: What do you expect the customer to put over here? Is "FooBar" valid? Can the descriptions be more meaningful with some example values.

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: General comment please provide required and readOnly properties of all the model definitions accurately. It is hard to believe that everything is optional and free flowing.

**Review by amarzavery — CHANGES_REQUESTED** [review]

**Comment by pankajsn** [comment]
@amarzavery
These are already existing operations, looks like RP (/providers/Microsoft.Logic/operations) is not updated with the full list of operations.
Our primary object of the current PR is to unblock few customers by fixing couple of PowerShell bugs. Including all these changes in the existing code, will take some extra time and we don’t want to miss the current release cycle of PowerShell (code complete by 13th feb). Please suggest how can we accommodate this.

**Comment by jhendrixMSFT** [comment]
@pankajsn If there's a tight deadline then perhaps it makes sense to update this PR (or close this one and create a new one, whichever is easier) with only the necessary bug fixes for PowerShell (IIRC it was just a few lines of changes), and do this larger refactoring/API additions in a separate PR.  Thoughts?

**Comment by pankajsn** [comment]
@Joel Hendrix<mailto:jhendrix@microsoft.com>
Other than bug fixes, we want to merge the older version of spec into the latest one. This is just to ensure that customer hits the latest api version. Other than that we fixed couple of linter warnings. Why don’t we pull this change-set as this is already reviewed. Please suggest.

For all the other refactoring related fixes, we can pull into a separate PR. Undoing will also take some extra effort at this point.

**Comment by pankajsn** [comment]
@Joel Hendrix<mailto:jhendrix@microsoft.com>
As discussed, we will fix all the comments as part of current PR.

@Amar Zavery<mailto:amzavery@microsoft.com>
Some of the comments below are not pointing to the correct line numbers hence not clear. Could you please point it correctly ?

From: Pankaj Singh Negi
Sent: Wednesday, February 8, 2017 11:12 AM
To: 'Azure/azure-rest-api-specs' <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Mention <mention@noreply.github.com>; Vinay Singh <vinaysin@microsoft.com>
Subject: RE: [Azure/azure-rest-api-specs] [Logic App Integration Account][Bug Fix] : Swagger spec updated as per the latest changes in the RP … (#891)

@amarzavery
These are already existing operations, looks like RP (/providers/Microsoft.Logic/operations) is not updated with the full list of operations.
Our primary object of the current PR is to unblock few customers by fixing couple of PowerShell bugs. Including all these changes in the existing code, will take some extra time and we don’t want to miss the current release cycle of PowerShell (code complete by 13th feb). Please suggest how can we accommodate this.

**Comment by pankajsn** [comment]
@Amar Zavery<mailto:amzavery@microsoft.com> @Joel Hendrix<mailto:jhendrix@microsoft.com>

Below highlighted comments are not pointing the correct location hence not very clear. Is there any way to update these ?

**Comment by amarzavery** [comment]
@pankajsn - I am sorry, cannot do anything about it. It is a github issue. As per github guidelines, it is expected that the author resolves all the comments made by the reviewer before making a new commit. With the new commit the line numbers change and nothing can be done about it.

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not required as described in RP definitions

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: comment not clear

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: free form JToken: user can any schema

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to find this one

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: Currently its a place holder (by design)

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed accross

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed notAfter

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to find this one

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed accross

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to find this one

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to locate this one

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to locate this one. But getting no linter warning

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: I was referring to this thing "content". What is the schema? Is it free form?

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: not able to locate this one. But getting no linter warning

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed accross

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: One more question: Is the properties required or optional?
I mean will the service send a 400 if the customer does not provide `properties: {}` in the request body

Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: fixed across

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: What is the qualifier? Is this an enum? How would the customer know what to provide?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Can you provide an example in the description of qualifier and value

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Are there any required properties in "AS2OneWayAgreement"?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: Are there any required properties in "AS2ProtocolSettings"? I mean if the user provides an empty object what would the service do? Accept it and apply defaults or send a 400?

Comment (amarzavery) [inline arm-logic/2016-06-01/swagger/logic.json]: same as above. The service is having huge model definitions. But everything is optional. It looks like garbage in garbage out. There is no way to know what should be provided on a bear minimum level

File: arm-logic/2016-06-01/swagger/logic.json
Comment (jhendrixMSFT) [inline arm-logic/2016-06-01/swagger/logic.json]: I don't see this marked as read-only, has that fix not been pushed yet?

File: arm-logic/2016-06-01/swagger/logic.json
Comment (jhendrixMSFT) [inline arm-logic/2016-06-01/swagger/logic.json]: Looks like this hasn't been addressed yet?

**Comment by pankajsn** [comment]
@jhendrixMSFT this is fixed in the latest iteration.

File: arm-logic/2016-06-01/swagger/logic.json
Comment (pankajsn) [inline arm-logic/2016-06-01/swagger/logic.json]: Part of new iteration

File: arm-logic/2016-06-01/swagger/logic.json
Comment (jhendrixMSFT) [inline arm-logic/2016-06-01/swagger/logic.json]: Looks like this value was already there.  @amarzavery did you mean a different enum?

**Comment by pankajsn** [comment]
@Amar Zavery<mailto:amzavery@microsoft.com> @Joel Hendrix<mailto:jhendrix@microsoft.com>

Can we please complete this PR ? Also,  please let me know if there are more comments.

**Comment by jhendrixMSFT** [comment]
@amarzavery I believe you're good with the current iteration, if so can you please sign off on this PR so we can merge it?

**Review by amarzavery — APPROVED** [review]

**Pull request merged**
